code/game/objects/effects/spawners/random/contraband.dm

```diff
@@ -24,6 +24,10 @@
 		/obj/effect/spawner/random/contraband/plus = 10,
 		/obj/item/reagent_containers/pill/maintenance = 5,
 	)
+	will_modify_loot = TRUE
+
+/obj/effect/spawner/random/contraband/modify_loot()
+	ADD_TRAIT(src, TRAIT_CONTRABAND, TRAIT_GENERIC) // No matter what, it's contraband, and should be treated as such.
```

Comment:
`INNATE_TRAIT`, also, you're applying the trait to the contraband spawner itself and not the loot